README.md
# NAME

Dist::Zilla::Plugin::InsertExample::FromMojoTemplates - Creates POD examples from a custom template format (deprecated)

![Requires Perl 5.10.1+](https://img.shields.io/badge/perl-5.10.1+-brightgreen.svg) [![Travis status](https://api.travis-ci.org/Csson/p5-Dist-Zilla-Plugin-InsertExample-FromMojoTemplates.svg?branch=master)](https://travis-ci.org/Csson/p5-Dist-Zilla-Plugin-InsertExample-FromMojoTemplates) 

# VERSION

Version 0.0301, released 2016-01-25.

# SYNOPSIS

    ; In dist.ini
    [InsertExample::FromMojoTemplates]
    directory = examples/source
    filepattern = ^\w+-\d+\.mojo$

# DESCRIPTION

**Deprecated**. See [Pod::Elemental::Transformer::Stenciller](https://metacpan.org/pod/Pod::Elemental::Transformer::Stenciller) instead.

Dist::Zilla::Plugin::InsertExample::FromMojoTemplates inserts examples from [MojoX::CustomTemplateFileParser](https://metacpan.org/pod/MojoX::CustomTemplateFileParser) type files into POD.
Together with [Dist::Zilla::Plugin::Test::CreateFromMojo](https://metacpan.org/pod/Dist::Zilla::Plugin::Test::CreateFromMojo) this produces examples in POD from the same source that creates the tests.
The purpose is to help develop tag helpers for [Mojolicious](https://metacpan.org/pod/Mojolicious).

## Attributes

**`directory`**

Default: `examples/source`

Where DZP::IE::FMT should look for source files.

**`filepattern`**

Default: `^\w+-\d+\.mojo$`

Look for files that matches a certain pattern.

**`make_examples`**

Default: `1`

If true, will create html files in the chosen directory.

**`example_directory`**

Default: `examples`

The directory for html files.

## USAGE

Source files looks like this:

    ==test example 1==
     --t--
         %= link_to 'The example 3' => ['http://www.perl.org/']
     --t--
     --e--
         <a href="http://www.perl.org/">Perl</a>
     --e--

This is a test block. One file can have many test blocks.

In your pod:

    # EXAMPLE: filename.mojo:1, 3-30, !5, !22-26

    # EXAMPLE: filename.mojo:all

    # EXAMPLE: filename.mojo:examples

**`all`**

Adds all examples in the source file. `all` can be used by itself or combined with exclusion commands.

**`1`**

Adds example number `3`. The test number is sequential. Looping tests count as one. You can add a number as in the example to make it easier to follow.

**`3-30`**

Add examples numbered `5` through `30`.

**`!5`**

Excludes example `5` from the previous range.

**`!22-26`**

Excludes examples numbered `22-26` from the previous range. If an example has been excluded it can't be included later. Exclusions are final.

**`examples`**

Includes all tests marked `==test example==` in the source file. Exclusion works as with `all`.

# SEE ALSO

The successor to this module is [Pod::Elemental::Transformer::Stenciller](https://metacpan.org/pod/Pod::Elemental::Transformer::Stenciller).

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-Plugin-InsertExample-FromMojoTemplates](https://github.com/Csson/p5-Dist-Zilla-Plugin-InsertExample-FromMojoTemplates)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-Plugin-InsertExample-FromMojoTemplates](https://metacpan.org/release/Dist-Zilla-Plugin-InsertExample-FromMojoTemplates)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-InsertExample-FromMojoTemplates

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Testing]

0.0301 2016-01-25T15:52:05Z UTC
 [Other Changes]
 - Deprecated.
 - Rebuilt with current Dist::Iller::Config::Author::CSSON.

 [Requirements]
 - namespace::sweep changed to namespace::autoclean.

0.0300 Unreleased
 - Version format change.

0.02.01 2014-09-18T21:22:43Z CEST
 [Meta]
 - Fix missing prereqs.

0.02.00 2014-09-10T19:05:54Z CEST
 [Documentation]
 - Fix name typos

 [Meta]
 - Now requires MojoX::CustomTemplateFileParser 0.09

 [New Features]
 - Support for including all tests marked as 'example' in the source file.
 - Two new attributes: make_examples (boolean) and example_directory
   (string)
 - These are used to create html files in the chosen folder

0.01.01 2014-09-09T13:01:50Z CEST
 [Meta]
 - Fix wrong dist name

0.01.00 2014-09-08T23:07:10Z CEST
 - Initial release

0.00.00 2014-09-08T14:55:39Z CEST
 - Before release



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;



use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Creates POD examples from a custom template format (deprecated)",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-InsertExample-FromMojoTemplates",
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::InsertExample::FromMojoTemplates",
  "PREREQ_PM" => {
    "Dist::Zilla" => "5.000",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "File::Find::Rule" => "0.33",
    "MojoX::CustomTemplateFileParser" => "0.09",
    "Moose" => "2.0000",
    "Path::Tiny" => "0.050",
    "namespace::autoclean" => "0.18"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0301",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "5.000",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Find::Rule" => "0.33",
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "MojoX::CustomTemplateFileParser" => "0.09",
  "Moose" => "2.0000",
  "Path::Tiny" => "0.050",
  "Test::More" => "0.96",
  "namespace::autoclean" => "0.18"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
